docs/bmo-part-of-capm3.md

```diff
@@ -0,0 +1,182 @@
+be able to smoothly migrate to the new layout (to generate new API versions?) as
+well as leaving only BMO deployments in deploy folder, while taking ironic
+deployments outside of the folder and moving them to the root of the repository
+to the new created `baremetal-operator/ironic-deployment` folder. The following
+directory tree visualize new structure of deploy folder in more detail:
+
+## Proposed structure of baremetal-operator deployment folder
+
+```diff
+tree deploy/
+
+deploy/
+├── crds
+│   ├── kustomization.yaml
+│   └── metal3.io_baremetalhosts_crd.yaml
+├── default
+│   ├── ironic_bmo_configmap.env
+│   └── kustomization.yaml
+├── ironic_ci.env
+├── namespace
+│   ├── kustomization.yaml
+│   └── namespace.yaml
+├── operator
+│   ├── bmo.yaml
+│   └── kustomization.yaml
+├── rbac
+│   ├── kustomization.yaml
+│   ├── role_binding.yaml
+│   ├── role.yaml
+│   └── service_account.yaml
+└── role.yaml -> rbac/role.yaml
+
+```
+
+As we can see, the deploy directory has one top level folder for deployment,
+namely `default` and it deploys only baremetal operator through kustomization.
+In addition, `crds`, `namespace` and `rbac` folders have their own kuztomization
+and yaml files. The following directory tree visualize a new structure of ironic
+-deployment folder in more detail:
+
+## Proposed structure of ironic-deployment folder
+
+```diff
+tree ironic-deployment/
+
+ironic-deployment/
+├── default
+│   ├── ironic_bmo_configmap.env
+│   └── kustomization.yaml
+├── ironic
+│   ├── ironic.yaml
+│   └── kustomization.yaml
+└── keepalived
+    ├── ironic_bmo_configmap.env
+    ├── keepalived_patch.yaml
+    └── kustomization.yaml
+
+```
+
+Whereas, ironic-deployment folder has three top level folder for deployments,
+namely `default`,  `ironic` and `keepalived`. `default` and `ironic` deploy
+only ironic, `keepalived` deploys the ironic with keepalived. As the name
+implies, `keepalived/keepalived_patch.yaml` patches the default image URL
+through kustomization.
+
+To this end, "production", "local developer" and "ironic outside of the
+cluster" use cases will be met, as new structure will fulfill the needs of those
+cases with cleaner and leaner organization of deployments in place.
+
+### Pivoting use case
+
+Once above outlined folder structural changes are made, we can deploy BMO on its
+own in CAPM3 and Ironic independently to meet the use case of pivoting, where
+BMO should deployed as part of CAPM3. The main concern here is that, BMO and
+Ironic are separated and as a consequence, it will result in breaking the chain
+of workflow, where, previously, BMO could use localhost to talk to ironic
+without any authentication. However, considering that Ironic runs in host
+network, BMO would be able to interact with it, from anywhere that is connected
+or routed to the provisioning network. Therefore, separating Ironic in its own
+pod should not impact BMO in this case.
```

Comment:
We will need a way to give BMO credentials for the Ironic API. I think we can set the URL now, but not the credentials.

Reply:
that can be done through clusterctl as a parameter for the deployment. or we could even use a secret to share it between ironic and BMO.